### PR TITLE
Send world spawn to players when necessary

### LINKED ISSUE #150: Send world spawn to players when necessary
Currently the world spawn isn't sent to the player after its initial spawn. The spawn position should be updated on world change and when the position of the world that the player is in is changed.

### PR DESCRIPTION
**Pull request by AndreasHGK**
Closes #150

### CODE REVIEW

**Comment by AndreasHGK** [comment]
Should I also add a method for this in the Player struct so people can update the spawn position for instance to point a compass somewhere? If so I'll do that otherwise you can review this PR

File: server/world/world.go [hunk 1]
```diff
@@ -1451,6 +1454,7 @@ func (w *World) addViewer(c *chunkData, viewer Viewer) {
 	}
 
 	viewer.ViewTime(w.Time())
+	viewer.ViewWorldSpawn(w.Spawn())
```
Comment (Sandertv) [inline server/world/world.go:1457]: You might want to move this to the end of `(*World*).addWorldViewer()`. The `ViewTime` call is actually misplaced too, as this function is called for every chunk loaded. Would you mind moving that to the same function too?
Thanks!

File: server/session/world.go [hunk 2]
```diff
@@ -744,6 +744,17 @@ func (s *Session) ViewEmote(player world.Entity, emote uuid.UUID) {
 	})
 }
 
+// ViewWorldSpawn ...
+func (s *Session) ViewWorldSpawn(pos cube.Pos) {
+	blockPos := protocol.BlockPos{int32(pos[0]), int32(pos[1]), int32(pos[2])}
+	s.writePacket(&packet.SetSpawnPosition{
+		SpawnType:     packet.SpawnTypeWorld,
+		Position:      blockPos,
+		Dimension:     1,
```
Comment (Sandertv) [inline server/session/world.go:753]: I think 1 is the nether, not sure if that's right? Maybe this works too though, I don't know.

Comment (AndreasHGK) [reply server/session/world.go]: You are right, I was looking at PM for this but I looked at the wrong file, woops 😅

File: server/world/world.go (hunk 1 above)
Comment (AndreasHGK) [reply server/world/world.go]: Done!

**Review by Sandertv — APPROVED** [review]
Looks good, thank you for the PR!

**Pull request merged**